src/Navigation/WPF/Navigation.WPF/Service/NavigationService.registration.cs

```diff
@@ -131,47 +164,59 @@ private static void RegisterInternal(INavigationRegistration registration)
 
             registrationInfo = NavigationRegistrations.FirstOrDefault(registration =>
                 registration.ViewModelType == viewModelType &&
-                viewType.IsAssignableFrom(registration.ViewType));
+                viewType.IsAssignableFrom(registration.ViewType) &&
+                registration.Condition?.Invoke() != false);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5575746)